backend/tests/integration/records/api/test_api_literature.py

```diff
@@ -1608,3 +1608,118 @@ def test_literature_updates_refs_to_known_and_unknown_conference_when_ref_alread
     lit = create_record("lit", data=lit_data)
 
     assert expected_publication_info == lit["publication_info"]
+
+
+def test_do_not_add_files_which_are_already_on_s3(
+    base_app, db, es, create_record, enable_files, s3
+):
+    data = {
+        "documents": [
+            {
+                "filename": "1905.03764.pdf",
+                "key": "some_document_key_on_s3",
+                "original_url": "http://inspire-afs-web.cern.ch/var/data/files/g188/3771224/content.pdf%3B2",
+                "source": "arxiv",
+                "url": "https://s3.cern.ch/inspire-qa-files-s/some_document_key_on_s3",
+            }
+        ],
+        "figures": [
+            {
+                "caption": "some caption",
+                "filename": "Global_noH3_EW_couplings_flat.png",
+                "key": "some_figure_key_on_s3",
+                "original_url": "http://inspire-afs-web.cern.ch/var/data/files/g188/3771220/content.png%3B2",
+                "source": "arxiv",
+                "url": "https://s3.cern.ch/inspire-qa-files-s/some_figure_key_on_s3",
+            }
+        ],
+    }
+    with mock.patch.object(
+        s3, "replace_file_metadata"
+    ) as mocked_s3_reploace_metadata_method:
+        record = create_record("lit", data=data)
+        mocked_s3_reploace_metadata_method.assert_not_called()
+    expected_documents = [
```

Comment:
aren't these the same as the initial documents and figures? You could avoid duplicating this

Reply:
Yes, you are right.